data/lib/core/container.lua

```diff
@@ -1,3 +1,35 @@
 function Container.isContainer(self)
 	return true
 end
+
+function Container.createLootContainer(self, lootblock)
+	if #lootblock == 0 then
+		return true
+	end
+	for i = 1, #lootblock do
+		if self:getSize() < self:getCapacity() then
+			local loot = lootblock[i]
+			local itemList = createLootItem(loot)
+			if (#itemList > 0) then
+				for i = 1, #itemList do
+					local item = itemList[i]
+					-- check containers
+					if item:isContainer() then
+						if not item:createLootContainer(loot.childLoot) then
```

Comment:
Same as above: In the original C++ code, the container item would be deleted from memory. You're just forgetting about it and leaving the object around creating a memory leak.